service/endpoint/internal/src/main/java/org/eclipse/kapua/service/endpoint/internal/EndpointInfoServiceImpl.java

```diff
@@ -183,19 +183,31 @@ public EndpointInfo find(KapuaId scopeId, KapuaId endpointInfoId)
 
         //
         // Check Access
-        EndpointInfo endpointInfoToFind = entityManagerSession.doAction(em -> EndpointInfoDAO.find(em, scopeId, endpointInfoId));
-        KapuaId scopeIdPermission = null;
-        if (endpointInfoToFind != null && endpointInfoToFind.getEndpointType().equals(EndpointInfo.ENDPOINT_TYPE_CORS)) {
-            scopeIdPermission = scopeId;
-        }
 
         AUTHORIZATION_SERVICE.checkPermission(
-                PERMISSION_FACTORY.newPermission(EndpointInfoDomains.ENDPOINT_INFO_DOMAIN, Actions.read, scopeIdPermission)
+                PERMISSION_FACTORY.newPermission(EndpointInfoDomains.ENDPOINT_INFO_DOMAIN, Actions.read, scopeId)
         );
 
-        //
-        // Do find
-        return entityManagerSession.doAction(em -> EndpointInfoDAO.find(em, scopeId, endpointInfoId));
+        EndpointInfo endpointInfoToFind = entityManagerSession.doAction(em -> EndpointInfoDAO.find(em, KapuaId.ANY, endpointInfoId)); // search the endpoint in any scope
+
+        if (endpointInfoToFind == null) {
+            throw new KapuaEntityNotFoundException(EndpointInfo.TYPE, scopeId);
```

Comment:
This should be:
`throw new KapuaEntityNotFoundException(EndpointInfo.TYPE, endpointInfoId);`